src/execution/StepExecutionManager.ts

```diff
@@ -43,27 +54,108 @@ export class StepExecutionManager {
-      !isZeroAddress(step.action.fromToken.address)
-    ) {
-      await checkAllowance(
+      !isZeroAddress(step.action.fromToken.address) &&
+      (shouldBatchTransactions || !isMultisigSigner)
+
+    if (checkForAllowance) {
+      const populatedTransaction = await checkAllowance(
         signer,
         step,
         statusManager,
         settings,
         fromChain,
-        this.allowUserInteraction
+        this.allowUserInteraction,
+        shouldBatchTransactions
       )
+
+      if (populatedTransaction) {
+        const { to, data } = populatedTransaction
+
+        if (to && data) {
+          // allowance doesn't need value
+          const cleanedPopulatedTransaction: BaseTransaction = {
+            value: '0x00',
+            to,
+            data,
+          }
+
+          multisigBatchTransactions.push(cleanedPopulatedTransaction)
+        }
+      }
     }
 
     // STEP 2: Get transaction
     let process = statusManager.findOrCreateProcess(step, currentProcessType)
 
     if (process.status !== 'DONE') {
+      const multisigProcess = step.execution.process.find(
+        (p) => !!p.multisigTxHash
+      )
+
       try {
-        let transaction: TransactionResponse
+        if (isMultisigSigner && multisigProcess) {
+          if (!multisigProcess) {
+            throw new Error('Multisig process is undefined')
+          }
+          if (!config.multisigConfig?.getMultisigTransactionDetails) {
+            throw new Error(
+              '"getMultisigTransactionDetails()" is missing in Multisig config.'
+            )
+          }
+
+          const multisigTxHash = multisigProcess.multisigTxHash
+
+          if (!multisigTxHash) {
+            // need to check what happens in failed tx
+            throw new Error('Multisig internal transaction hash is undefined.')
+          }
+
+          const response: MultisigTxDetails =
+            await config.multisigConfig?.getMultisigTransactionDetails(
+              multisigTxHash
+            )
+
+          if (response.status === 'PENDING') {
+            return step.execution!
+          }
+
+          if (response.status === 'SUCCESS') {
+            process = statusManager.updateProcess(
+              step,
+              process.type,
+              'PENDING',
+              {
+                txHash: response.txHash,
+                multisigTxHash: undefined,
+                txLink:
+                  fromChain.metamask.blockExplorerUrls[0] +
+                  'tx/' +
+                  response.txHash,
+              }
+            )
+          }
+
+          if (response.status === 'FAILED') {
+            throw new TransactionError(
+              LifiErrorCode.TransactionFailed,
+              'Multisig transaction failed.'
+            )
+          }
+
+          if (response.status === 'CANCELLED') {
+            throw new TransactionError(
+              LifiErrorCode.TransactionRejected,
+              'Transaction was rejected by users'
```

Comment:
Let's be consistent with messages and dots at the end.

Reply:
alright!
Removing the `.` from everywhere